### PR TITLE
Update UIRichText

### PR DESCRIPTION
**Pull request by LingJiJian**
Update UIRichText  underline outline  cache label/sprite

### CODE REVIEW

File: cocos/ui/UIRichText.h [hunk 1]
```diff
@@ -102,29 +126,45 @@ class RichText : public Widget
...
+    virtual void adaptRenderers();
+
     virtual void initRenderer();
     void pushToContainer(Node* renderer);
-    void handleTextRenderer(const std::string& text, const std::string& fontName, float fontSize, const Color3B& color, GLubyte opacity);
-    void handleImageRenderer(const std::string& fileParh, const Color3B& color, GLubyte opacity);
-    void handleCustomRenderer(Node* renderer);
+    void handleTextRenderer(const std::string& text, const std::string& fontName, float fontSize,bool isUnder, const Color3B& color, GLubyte opacity,int tag,bool isOutLine,const Color4B& outLineColor);
+    void handleImageRenderer(const std::string& fileParh, const Color3B& color, GLubyte opacity,int tag,bool isAnim,float delay,bool isLoop);
+    void handleCustomRenderer(Node* renderer,int tag);
```
Comment (minggo) [inline cocos/ui/UIRichText.h:153]: Will break compatibility?

**Comment by minggo** [comment]
Can i close #11280 and #11281?

**Comment by LingJiJian** [comment]
可以

File: cocos/ui/UIRichText.h (hunk 1 above)
Comment (LingJiJian) [inline cocos/ui/UIRichText.h:153]: 兼容的   compatibility

Comment (minggo) [inline cocos/ui/UIRichText.h:153]: But the interfaces are changed.

Comment (LingJiJian) [inline cocos/ui/UIRichText.h:153]: changed's interfaces is protected   is ok ..

**Comment by zilongshanren** [comment]
@LingJiJian 
This PR contains merge conflicts now. Could you resolve it?
这个PR有冲突了，你可以解决一下吗？

**Comment by LingJiJian** [comment]
- \-  我本地没有clone的........我不知道在哪可以查看冲突文件.....

**Comment by minggo** [comment]
@LingJiJian 
How did you send a pull request if you don't have a cloned repo?

**Comment by LingJiJian** [comment]
i just use web .. no git client      by the way   the RichText i edit with cocoswidget    https://github.com/LingJiJian/Tui-x/blob/master/CocosWidget/TextRich.cpp

**Pull request closed without merging**